MediaProcessor/src/AudioProcessor.cpp

```diff
@@ -15,19 +15,40 @@ AudioProcessor::AudioProcessor(const std::string &inputVideoPath,
                                const std::string &outputAudioPath)
     : m_inputVideoPath(inputVideoPath),
       m_outputAudioPath(outputAudioPath),
-      m_numChunks(DEFAULT_NUM_CHUNKS),
+      m_numChunks(determineNumChunks()),
```

Comment:
Please remove this from the member initialization. Instead, initialize it inside the constructor body by using the `ConfigManager` singleton via the new `getOptimalThreadCount`